rizinorg/rizin
# RzUtil: fix rz_path_prefix in case of portable build

**Your checklist for this pull request**

**Detailed description**



Before this patch the function was using RZ_PREFIX even when IS_PORTABLE
was set, making the function not working well because RZ_PREFIX might
not exist in the system where the portable build is used (the path was
passed to realpath(), which requires an existing path).

This patch fixes the problem by searching for the BINDIR without using
realpath() on the RZ_PREFIX. Moreover, it avoid retrieving the same
prefix every time by saving it in a global variable.


**Test plan**



- Build with -Dportable=true, `eco` should list all themes
- Build with -Dportable=true --bindir=., `eco` should list all themes
- Build with -Dportable=false, `eco` should list all themes

To better test, spawn a clean container so that files are not loaded from user home dir or other system places.

@wargio Would this work for Cutter as well?
